runtime/Python2/src/antlr4/Recognizer.py

```diff
@@ -45,14 +45,20 @@ def removeErrorListener(self, listener):
     def removeErrorListeners(self):
         self._listeners = []
 
+    def getTokenNames(self):
+        return tuple(self.literalNames + self.symbolicNames[len(self.literalNames):])
+
+    def getRuleNames(self):
+        return tuple(self.ruleNames)
```

Comment:
same here

Reply:
and why not just use self.ruleNames ?

Reply:
Defining the getRuleNames function allows better backwards compatibility for anyone overriding it, rather than just replacing the one call site. Although my change to return a dict from getRuleIndexMap may already be a breaking change... let me know what you'd prefer.

Reply:
since the function doesn't currently exist, not sure what you mean by "backwards compatibility " ?

Reply:
Users of the library can still define it, supposing that getRuleIndexMap is functioning correctly otherwise.